### PR TITLE
Write all tags in saved images

### LINKED ISSUE #39: Figure out how to build an image with multiple tags and push it
Without building multiple images.

### PR DESCRIPTION
**Pull request by La0**
`img` does not write all the tags for an image, so we need to patch the produced image's manifest.

Fixes #39 & #42

### CODE REVIEW

File: taskboot/build.py [hunk 1]
```diff
@@ -147,9 +146,7 @@ def build_compose(target, args):
 
         # Write the produced image
         if output:
-            for tag in tags:
-                output_file = '{}.tar'.format(tag.replace(os.sep, "_").replace(":", "_"))
-                docker.save(tag, os.path.join(output, output_file))
+            docker.save(tags, os.path.join(output, '{}.tar'.format(service)))
```
Comment (marco-c) [inline taskboot/build.py]: This used to be `name` before https://github.com/mozilla/task-boot/pull/38, maybe we want to keep that?

File: taskboot/docker.py [hunk 2]
```diff
@@ -245,3 +256,37 @@ def _find_replacement(original):
...
+
+
+def read_manifest(path):
+    '''
+    Read a Docker archive manifest and load it as JSON
+    '''
+    assert os.path.exists(path), 'Missing archive {}'.format(path)
+    assert tarfile.is_tarfile(path), 'Not a TAR archive {}'.format(path)
+
+    tar = tarfile.open(path)
```
Comment (marco-c) [inline taskboot/docker.py]: Nit: you could use a context manager here so you ensure the tar file is closed after.

File: taskboot/docker.py [hunk 3]
```diff
@@ -245,3 +256,37 @@ def _find_replacement(original):
...
+    manifest_raw = tar.extractfile('manifest.json')
+    return json.loads(manifest_raw.read().decode('utf-8'))
```
Comment (marco-c) [inline taskboot/docker.py]: Curiosity more than anything else, can't json.load be used here with the file object returned by extractfile?

File: taskboot/docker.py [hunk 4]
```diff
@@ -245,3 +256,37 @@ def _find_replacement(original):
...
+
+    # Tar file content must be provided as bytes
+    content = json.dumps(manifest).encode('utf-8')
+
+    # TarInfo is the pointer to the data in the tar archive
+    index = tarfile.TarInfo('manifest.json')
+    index.size = len(content)
+
+    # Open the archive in append mode, and overwrite the existing file
+    tar = tarfile.open(path, 'a')
```
Comment (marco-c) [inline taskboot/docker.py]: Nit: same here, context manager

File: tests/conftest.py [hunk 5]
```diff
@@ -29,3 +32,14 @@ def run(self, command, **kwargs):
                 raise Exception('Unsupported command in mock: {}'.format(' '.join(command)))
 
     return MockDocker()
+
+
+@pytest.fixture
+def hello_archive(tmpdir):
```
Comment (marco-c) [inline tests/conftest.py]: Nit: you could use `tmp_path`, which is the newer version of the temporary directory fixture

File: taskboot/build.py (hunk 1 above)
Comment (La0) [reply taskboot/build.py]: :man_facepalming:

**Pull request merged**